test_framework/basic.py

```diff
@@ -310,22 +311,27 @@ def validate_runs(
         expected_retcode = expected["return_code"]
         expected_stdout = expected.get("stdout", "")
 
+        def build_error_message_wrapped():
```

Comment:
Not sure if that's really necessary, but instead of adding the `source_file` three times I decided to create this closure :)